code/ecdsap256sha2/Hacl.Impl.LowLevel.fst

```diff
@@ -701,151 +920,179 @@ let lemma_320_64 a b c d e u =
 
 
 val sq0: f:  lbuffer uint64 (size 4) -> result: lbuffer uint64 (size 4) -> memory: lbuffer uint64 (size 12) -> temp: lbuffer uint64 (size 5) -> Stack uint64
-  (requires fun h -> live h result /\ live h f /\ live h memory /\ live h temp /\ disjoint result temp
-    /\ disjoint result memory /\ disjoint memory temp 
- )
+  (requires fun h -> live h result /\ live h f /\ live h memory /\ live h temp /\ 
+    disjoint result temp /\ disjoint result memory /\ disjoint memory temp 
+  )
   (ensures fun h0 c h1 -> modifies (loc result |+| loc memory |+| loc temp) h0 h1 /\ 
     (
       let f0 = Lib.Sequence.index (as_seq h0 f) 0 in 
-      as_nat h1 result + uint_v c * pow2 64 * pow2 64 * pow2 64 * pow2 64  = uint_v f0 * as_nat h0 f
-
-    ) /\
-
+      as_nat h1 result + uint_v c * pow2 64 * pow2 64 * pow2 64 * pow2 64  = uint_v f0 * as_nat h0 f) /\
+      (
+  let f = as_seq h0 f in 
+  let f0 = Lib.Sequence.index f 0 in 
```

Comment:
for this and all other occurrences below, I would suggest defining helpers that return a tuple, for readability